04.class/operateInterface.js

```diff
@@ -0,0 +1,97 @@
+import readline from "readline";
+import enquirer from "enquirer";
+
+const { Select } = enquirer;
+export default class OperateInterface {
+  constructor(memoOperation) {
+    this.memoOperation = memoOperation;
+  }
+
+  async listMemos() {
+    try {
+      const rows = await this.memoOperation.fetchMemos();
+      rows.forEach((row) => {
+        console.log(row.content.split("\n")[0]);
+      });
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  async readMemos() {
+    try {
+      const result = await this.#selectMemosFromList("see");
+      console.log(result);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  async deleteMemo() {
+    try {
+      const result = await this.#selectMemosFromList("delete");
+      this.memoOperation.removeMemo(result);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  async writeMemo() {
+    try {
+      const content = await this.#createMemoPrompt();
+      await this.memoOperation.addMemo(content);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  async #selectMemosFromList(action) {
+    try {
+      const rows = await this.memoOperation.fetchMemos();
+      const memos = this.#extractFirstLineAndValue(rows, action);
+      return this.#promptForMemo(memos, action);
+    } catch (err) {
+      console.error(err);
+    }
+  }
+
+  #extractFirstLineAndValue(rows, action) {
+    return rows.map((row) => {
+      const firstLine = row.content.split("\n")[0] || " ";
```

Comment:
```suggestion
      const firstLine = row.content.split("\n")[0] || "";
```

細かいんですが、空文字列じゃなくてスペースにしないといけない理由って何かありましたか？

Reply:
特に理由はないのですが、空文字で実行した際にエラーが発生したと思いスペースを入れた気がします。
ただ、別の原因でエラーが発生した可能性が高いので、空文字へと修正いたします。

Reply:
追記：
今、空文字で修正してから実行をしたら、挙動が変わったことで思い出しました。
空文字にすると、`-r`時に`id`が数値で表示されたり、`-r`時に`content`も改行ありで表示されるようになったため、スペースを入れました。
空文字での修正ができないか考えたのですが、思いつかなかったのでスペースを入れた形で再度提出いたします。